client/pluginManger.go

```diff
@@ -133,6 +139,53 @@ func (p *PluginManager) InstallPlugin(names []string) (err error) {
 	return
 }
 
+func (p *PluginManager) writeFileLocally(statusCode int, body []byte, name string) (err error) {
+	if statusCode != 200 {
+		err = fmt.Errorf("unexpected status code: %d", statusCode)
+		if p.Debug {
+			ioutil.WriteFile(debugLogFile, body, 0664)
+		}
+		return err
+	}
+	
+	if err = ioutil.WriteFile(fmt.Sprintf("%s.hpi", name), body, 0644); err != nil {
+		if p.Debug {
+			ioutil.WriteFile(debugLogFile, []byte(err.Error()), 0664)
+		}
+	}
+	return err
+}
+
+// InstallPluginWithVersion install a plugin by name & version
+func (p *PluginManager) installPluginWithVersion(name string) (err error) {
+	pluginName := "%s.hpi"
+	pluginVersion := strings.Split(name, "@")
+	url := fmt.Sprintf("http://updates.jenkins-ci.org/download/plugins/%s/%s/%s.hpi",
+		pluginVersion[0], pluginVersion[1], pluginVersion[0])
+
+	if resp, err := http.Get(url); err != nil {
+		fmt.Println(err)
```

Comment:
If there're some errors happen. I think it cannot keep going.  We probably should just return the error.